1521 error reporting for api

susanodd opened the pull request:
IMPORTANT: MERGE THIS BRANCH FIRST!

The Line Numbers in the PO file get totally messed up if other branches get merged "before" this one. 

This branch will need to work together with #1503 in order to make use of the translation languages in the request in Swagger.
The error messages can be viewed by using curl and the desired language code. This branch is on signbank-test.

The following error illustrated in the details below has been taken care of in #1497, #1503.

<details>

Because the retrieval urls for get_gloss_data work for both a logged in Signbank user as well as an API user, the language code is "retrieved" in different ways.

The language code extracted from the request "as it is now on master" does not always retrieve the correct language from the request because it depends on the browser. Firefox sends a huge string as Accept-Language when this is not done inside a proper API request. This causes it to not match anything and default to "en".

Firefox is doing this for Signbank (when the user is logged in):

This code:

`interface_language_code = request.headers.get('Accept-Language', 'en')`

retrieves this:

`nl,en-US;q=0.7,en;q=0.3`

I figured out how to solve this, but since this is the mechanics of making the translations be activated for the language, it is going to be done in the other issue, to avoid duplication.

Found this approach when searching:

<details>

```
from django.utils.translation.trans_real import parse_accept_lang_header
header_locales =parse_accept_lang_header(
                                  request.META.get('HTTP_ACCEPT_LANGUAGE', None))
```
Given example:

`[('pt-pt', 1.0), ('pt-br', 0.8), ('en-us', 0.5), ('en', 0.3)]`

vanlummelhuizen reviewed (changes requested):
Good to see more messages and more translation. This will help users a lot.

Some minor things that can be changed easily. Also I'd like to see why a check is removed (see my comments).

susanodd commented on conf/locale/nl/LC_MESSAGES/django.po:
```diff
@@ -1459,204 +1578,264 @@ msgstr "Vul nieuwe aantekening in"
+#: signbank/dictionary/gloss_revision.py:110
+#: signbank/dictionary/templates/dictionary/glosstags.html:22
+#: signbank/dictionary/templates/dictionary/morphemetags.html:22
+msgid "delete this tag"
```
Okay, but that needs to be in the templates.....

vanlummelhuizen replied:
Your are right, of course.

susanodd commented on conf/locale/nl/LC_MESSAGES/django.po:
```diff
@@ -338,78 +359,166 @@ msgstr "Aantekeningen"
+msgstr "Glos heeft geen dataset"
+
+#: signbank/api_interface.py:674
+msgid "No change gloss permission"
```
I need to change the original text in the templates, not in the po-file.

I'm the person who wrote the original. I guess because I'm used to computer jargon it probably sounds weird in my native language too, unfortunately.

susanodd commented:
I'm going to push the repairs done so far, then modify the template text you mention, then repeat.

susanodd commented on conf/locale/nl/LC_MESSAGES/django.po:
```diff
@@ -338,78 +359,166 @@ msgstr "Aantekeningen"
+msgstr "Geen verander-glos toestemming"
+
+#: signbank/api_interface.py:679
+msgid "No change permission for dataset"
```
Man, there are lots of these original ones for English. I didn't notice that the English sounded weird. I'm fixing them all.

susanodd commented:
> Good to see more messages and more translation. This will help users a lot.
> 
> Some minor things that can be changed easily. Also I'd like to see why a check is removed (see my comments).

Okay, all the text translations have been resolved. The original English text has been modified as requested in the templates and Python error messages.

I don't see the removed check comment?

vanlummelhuizen commented:
> This does not work as intended because Django doesn't match the url if it's not a digit sequence. (That's why I removed the code in the first place.)

This is the explanation I was looking for. I did not think of it. So, in all places where this applies, please remove the check, because it is redundant.

susanodd commented:
> > This does not work as intended because Django doesn't match the url if it's not a digit sequence. (That's why I removed the code in the first place.)
> 
> This is the explanation I was looking for. I did not think of it. So, in all places where this applies, please remove the check, because it is redundant.

I'm going to make a new issue for this. It also applies to other API urls that were not originally part of this issue.

Since it wasn't clear to me whether I removed a check that I should have kept, I put it back, and then made it so all the other code where it was missing also checks that the id is a number. The goal of this issue is the error reporting. So I added status codes where they were missing and text and gettext where it was missing.
